libusb/os/darwin_usb.c

```diff
@@ -481,15 +549,15 @@ static usb_device_t **darwin_device_from_service (struct libusb_context *ctx, io
 
   if (kIOReturnSuccess != kresult || !plugInInterface) {
     usbi_dbg (ctx, "could not set up plugin for service: %s", darwin_error_str (kresult));
-    return NULL;
+    return darwin_to_libusb(kresult);
   }
```

Comment:
This looks like an independent fix so should also be mentioned in the commit message.